sympy/utilities/iterables.py

```diff
@@ -523,7 +521,18 @@ def ibin(n, bits=0, str=False):
     ['000', '001', '010', '011', '100', '101', '110', '111']
 
     """
+    if n < 0:
+        raise ValueError(
+                "Negetive numbers are not allowed")
+    x, y = abs(n), []
+    y.append(x)
+    #y.reverse()
+
     if not str:
+        if (len(y) - 1) is not None:
```

Comment:
this test should be done before the `if`; please see how `n.bitlength()` can be used to make an efficient test. (And, of course, the error message will have to be corrected.)